crates/assists/src/handlers/unmerge_use.rs

```diff
@@ -0,0 +1,213 @@
+use syntax::{
+    algo::SyntaxRewriter,
+    ast::{self, edit::AstNodeEdit, VisibilityOwner},
+    AstNode, SyntaxKind,
+};
+
+use crate::{
+    assist_context::{AssistContext, Assists},
+    AssistId, AssistKind,
+};
+
+// Assist: unmerge_use
+//
+// Extracts single use item from use list.
+//
+// ```
+// use std::fmt::{Debug, Display$0};
+// ```
+// ->
+// ```
+// use std::fmt::{Debug};
+// use std::fmt::Display;
+// ```
+pub(crate) fn unmerge_use(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let tree: ast::UseTree = ctx.find_node_at_offset()?;
+
+    let tree_list = tree.syntax().parent().and_then(ast::UseTreeList::cast)?;
+    if tree_list.use_trees().count() < 2 {
+        return None;
+    }
+
+    let use_: ast::Use = tree_list.syntax().ancestors().find_map(ast::Use::cast)?;
+    let path = resolve_full_path(&tree)?;
+
+    let new_use = ast::make::use_(
+        use_.visibility(),
+        ast::make::use_tree(path, None, tree.rename(), tree.star_token().is_some()),
+    );
+
+    let mut rewriter = SyntaxRewriter::default();
+    rewriter += tree.remove();
+    rewriter.insert_after(use_.syntax(), &ast::make::tokens::single_newline());
+    if let ident_level @ 1..=usize::MAX = use_.indent_level().0 as usize {
+        rewriter.insert_after(
+            use_.syntax(),
+            &ast::make::tokens::whitespace(&" ".repeat(4 * ident_level)),
+        );
+    }
+    rewriter.insert_after(use_.syntax(), new_use.syntax());
```

Comment:
This part should go into the `|builder|` closure, to avoid unnecessary computation.